### ISSUE TITLE
✅ 📊 🐞 Bug Report: Elements CSS appear inline after updated to v3.16 if "Improved CSS Loading" is active, and site has cache - [ED-12212]

### ISSUE THREAD

**Issue by scferg** [elementor/elementor#23783]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

Elementor auto-updated to v3.16.2 and all of a sudden certain modules in a saved template are displaying CSS style literals on the page. They don't appear in the editor, only on the front end.

It seems that it's outputting the CSS on the page for the first occurrence of each Heading, Spacer, and Text Editor module (those are at least the ones being used) within the layout. No matter where I move the modules or recreate them from scratch, the first one of each in the layout is prepended with the CSS.

**CSS before Text Editor module**
```css
/*! elementor - v3.16.0 - 13-09-2023 */ .elementor-widget-text-editor.elementor-drop-cap-view-stacked .elementor-drop-cap{background-color:#69727d;color:#fff}.elementor-widget-text-editor.elementor-drop-cap-view-framed .elementor-drop-cap{color:#69727d;border:3px solid;background-color:transparent}.elementor-widget-text-editor:not(.elementor-drop-cap-view-default) .elementor-drop-cap{margin-top:8px}.elementor-widget-text-editor:not(.elementor-drop-cap-view-default) .elementor-drop-cap-letter{width:1em;height:1em}.elementor-widget-text-editor .elementor-drop-cap{float:left;text-align:center;line-height:1;font-size:50px}.elementor-widget-text-editor .elementor-drop-cap-letter{display:inline-block}
```

**CSS before Spacer module**
```css
/*! elementor - v3.16.0 - 13-09-2023 */ .elementor-column .elementor-spacer-inner{height:var(--spacer-size)}.e-con{--container-widget-width:100%}.e-con-inner>.elementor-widget-spacer,.e-con>.elementor-widget-spacer{width:var(--container-widget-width,var(--spacer-size));--align-self:var(--container-widget-align-self,initial);--flex-shrink:0}.e-con-inner>.elementor-widget-spacer>.elementor-widget-container,.e-con>.elementor-widget-spacer>.elementor-widget-container{height:100%;width:100%}.e-con-inner>.elementor-widget-spacer>.elementor-widget-container>.elementor-spacer,.e-con>.elementor-widget-spacer>.elementor-widget-container>.elementor-spacer{height:100%}.e-con-inner>.elementor-widget-spacer>.elementor-widget-container>.elementor-spacer>.elementor-spacer-inner,.e-con>.elementor-widget-spacer>.elementor-widget-container>.elementor-spacer>.elementor-spacer-inner{height:var(--container-widget-height,var(--spacer-size))}.e-con-inner>.elementor-widget-spacer.elementor-widget-empty,.e-con>.elementor-widget-spacer.elementor-widget-empty{position:relative;min-height:22px;min-width:22px}.e-con-inner>.elementor-widget-spacer.elementor-widget-empty .elementor-widget-empty-icon,.e-con>.elementor-widget-spacer.elementor-widget-empty .elementor-widget-empty-icon{position:absolute;top:0;bottom:0;left:0;right:0;margin:auto;padding:0;width:22px;height:22px}
```

**CSS before Heading module**
```css
/*! elementor - v3.16.0 - 13-09-2023 */ .elementor-heading-title{padding:0;margin:0;line-height:1}.elementor-widget-heading .elementor-heading-title[class*=elementor-size-]>a{color:inherit;font-size:inherit;line-height:inherit}.elementor-widget-heading .elementor-heading-title.elementor-size-small{font-size:15px}.elementor-widget-heading .elementor-heading-title.elementor-size-medium{font-size:19px}.elementor-widget-heading .elementor-heading-title.elementor-size-large{font-size:29px}.elementor-widget-heading .elementor-heading-title.elementor-size-xl{font-size:39px}.elementor-widget-heading .elementor-heading-title.elementor-size-xxl{font-size:59px}
```

Here is an example of the Text Editor module
![Screenshot 2023-09-13 at 9 26 29 PM](https://github.com/elementor/elementor/assets/1112334/c7b56bc9-30bf-4028-8e20-40520afd65b7)

I tried downgrading to a previous version, but that broke other things on the site, such as making all the modules on the Home page disappear.

### Steps to reproduce

1) Add a Heading, Spacer, and Text Editor module to a layout
2) No need to alter any of the styles or settings
3) Save the layout
4) View on the frontend and the CSS styles will be printed on the page

You can move the modules around or create new ones and the CSS styles seem to only get printed on the first occurrence of those module types in the layout.

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt


== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: Source distribution v5.7.39-42
	PHP Version: 7.4.33
	PHP Memory Limit: 768M
	PHP Max Input Vars: 3000
	PHP Max Post Size: 256M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3.1
	Site URL: https://url-redacted.com
	Home URL: https://url-redacted.com
	WP Multisite: No
	Max Upload Size: 256 MB
	Memory limit: 256M
	Max Memory limit: 768M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Admin Email: info@url-redacted.com
	Debug Mode: Inactive

== Theme ==
	Name: Loveus child
	Version: 1.6
	Author: smartdatasoft
	Child Theme: Yes
	Parent Theme Name: loveus
	Parent Theme Version: 1.8
	Parent Theme Author: smartdatasoft

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36

== Active Plugins ==
	Akismet Anti-spam: Spam Protection
		Version: 5.3
		Author: Automattic - Anti-spam Team

	Contact Form 7
		Version: 5.8
		Author: Takayuki Miyoshi

	Contact Form CFDB7
		Version: 1.2.6.7
		Author: Arshid

	Elementor
		Version: 3.16.2
		Author: Elementor.com

	List Custom Taxonomy Widget
		Version: 4.1
		Author: Nick Halsey

	LoveUs Core
		Version: 2.1
		Author: smartdatasoft

	Loveus Demo Installer
		Version: 1.3
		Author: ProteusThemes

	Meta Box
		Version: 5.7.5
		Author: MetaBox.io

	Redux Framework
		Version: 4.4.6
		Author: Team Redux

	SiteGround Central
		Version: 3.0.1
		Author: SiteGround

	SiteGround Security
		Version: 1.4.5
		Author: SiteGround

	Site Kit by Google
		Version: 1.109.0
		Author: Google

	The Events Calendar
		Version: 6.2.2
		Author: The Events Calendar

	WooCommerce
		Version: 8.1.0
		Author: Automattic

	Wordfence Security
		Version: 7.10.3
		Author: Wordfence

	WPFront Scroll Top
		Version: 2.1.1
		Author: WPFront Team


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Inactive by default
	Upgrade Swiper Library: Inactive by default
	Grid Container: Inactive by default
	Editor Top Bar: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Inactive by default


== Log ==
	
Log: showing 20 of 412023-09-12 16:05:18 [info] Elementor/Upgrades - _v_3_16_0_container_updates Start  
2023-09-12 16:05:18 [info] Elementor/Upgrades - _v_3_16_0_container_updates Finished 
2023-09-12 16:05:18 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-09-12 16:05:18 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-09-12 16:05:18 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.3',
  'to' => '3.16.0',
)]
2023-09-12 16:05:18 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.15.3',
  'to' => '3.16.0',
)]
2023-09-13 04:18:37 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.0',
  'to' => '3.16.1',
)]
2023-09-13 04:18:38 [info] elementor::elementor_updater Started 
2023-09-13 04:18:38 [info] Elementor/Upgrades - _on_each_version Start  
2023-09-13 04:18:38 [info] Elementor/Upgrades - _on_each_version Finished 
2023-09-13 04:18:38 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-09-13 04:18:38 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-09-13 04:18:38 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.0',
  'to' => '3.16.1',
)]
2023-09-13 12:05:32 [info] elementor::elementor_updater Started 
2023-09-13 12:05:32 [info] Elementor/Upgrades - _on_each_version Start  
2023-09-13 12:05:32 [info] Elementor/Upgrades - _on_each_version Finished 
2023-09-13 12:05:32 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Start  
2023-09-13 12:05:32 [info] Elementor/Upgrades - _v_3_17_0_site_settings_updates Finished 
2023-09-13 12:05:32 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.1',
  'to' => '3.16.2',
)]
2023-09-13 12:05:32 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.16.1',
  'to' => '3.16.2',
)]

JS: showing 2 of 2JS: 2023-09-14 00:27:19 [error X 10][https://url-redacted.com/wp-admin/load-scripts.php?c=0&loadchunk_0=jquery-core,jquery-migrate,utils,moxiejs,plupload&ver=6.3.1:2:28722] Cannot read properties of undefined (reading 'value') 
JS: 2023-09-14 01:01:51 [error X 1][https://url-redacted.com/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.16.2:3:54034] Cannot read properties of undefined (reading 'activate') 



== Elementor - Compatibility Tag ==

````

</details>

**Comment by avivu** [comment]
Hi @scferg,

Can you please try to deactivate the "Improved CSS loading" feature under WordPress > Elementor > Settings and check if it's solving the issue?

**Comment by scferg** [comment]
@avivu Awesome, that did the trick. Thank you! Does that setting get turned on by default in the latest Elementor version?

**Comment by avivu** [comment]
Yes @scferg, it was turned on to improve site performance and to allow us to create a better experience.
It seems we have some glitches we need to fix, but I'm glad it solved your problem.

**Comment by nicholaszein** [comment]
Hi @scferg!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Core v3.16.3! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
